src/Pelagos/Bundle/AppBundle/Controller/UI/TombstoneController.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Controller\UI;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
+use Pelagos\Entity\Dataset;
+use Pelagos\Entity\DatasetSubmission;
+
+/**
+ * The Dataset Tombstone controller.
+ *
+ * @Route("/tombstone")
+ */
+class TombstoneController extends UIController
+{
+    /**
+     * The Dataland Page - dataset details per UDI.
+     *
+     * @param string $udi A UDI.
+     *
+     * @throws NotFoundHttpException When no non-available dataset is found with this UDI.
+     *
+     * @Route("/{udi}")
+     *
+     * @return Response
+     */
+    public function defaultAction($udi)
+    {
+        $dataset = $this->getDataset($udi);
+
+        if (($dataset->getAvailabilityStatus() == DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE) or
+            ($dataset->getAvailabilityStatus() == DatasetSubmission::AVAILABILITY_STATUS_PUBLICLY_AVAILABLE_REMOTELY_HOSTED)) {
+            throw $this->createNotFoundException("No pending state placeholder found for UDI: $udi");
+        } else {
```

Comment:
This else is not really needed right?

Reply:
I guess not as the exception being thrown stops execution, but I left it in for clarity.

Reply:
I've since simplified it.